### ISSUE TITLE
avaje validations inside spring-boot runtime

### ISSUE THREAD

**Issue by jeusdi** [avaje/avaje-validator#199]
It's not clear to me, how method validaton works.

I mean, I'm playing with avaje-validations inside my spring boot service.

I'm aware of that avaje ecosystem offers injection and aop, but my purpose is trying to get working avaje validations inside a spring-boot runtime.

Here my related code:

```java
import org.springframework.stereotype.Component;

import io.avaje.validation.ValidMethod;
import io.avaje.validation.Validator;
import io.avaje.validation.constraints.NotNull;

@Component
public class DummyService {

  @ValidMethod
  public String dummyMethod(@NotNull String name) {
      return "Hello, " + name + "!";
  }

}
```

Here my generated code:

```java
import com.example.demo.service.DummyService;
import io.avaje.validation.adapter.MethodAdapterProvider;
import io.avaje.validation.adapter.ValidationAdapter;
import io.avaje.validation.adapter.ValidationContext;
import io.avaje.validation.constraints.NotNull;
import io.avaje.validation.spi.Generated;
import jakarta.inject.Singleton;
import java.lang.reflect.Method;
import java.util.List;
import java.util.Map;
import java.util.Set;

@Generated
@Singleton
public final class DummyMethodParamProvider implements MethodAdapterProvider {

  @Override
  public Method method() throws Exception {
    return DummyService.class.getDeclaredMethod("dummyMethod", String.class);
  }

  @Override
  public List<ValidationAdapter<Object>> paramAdapters(ValidationContext ctx) {
    return List.of(
        ctx.<Object>adapter(NotNull.class, Map.of("message","{avaje.NotNull.message}", "_type","String")));
  }

  @Override
  public ValidationAdapter<Object> returnAdapter(ValidationContext ctx) {
    return ctx.<Object>noop();
  }

  @Override
  public ValidationAdapter<Object[]> crossParamAdapter(ValidationContext ctx) {
    return ctx.<Object[]>noop();
  }
}
```

It's not clear to me how to validate my method parameters.
Note: By default spring is not using JSR-330 by default, bit I've set it up and my spring boot service [is able to wire JSR -330 annotations](https://docs.spring.io/spring-framework/reference/core/beans/standard-annotations.html).

Cloud you give some lights?

**Comment by SentryMan** [comment]
It's possible, basically you need to create a spring version of [this](https://github.com/avaje/avaje-validator/tree/main/validator-inject-plugin/src/main).

**Comment by SentryMan** [comment]
I can probably add another module for spring method validation so people don't have to implement manually

**Comment by rbygrave** [comment]
Released version 1.5-RC1, with io.avaje:avaje-validator-spring-starter:1.5-RC1 so you can try that.

**Comment by jeusdi** [comment]
Spring AOP works well with `@ValidMethod`.

Thanks!
